### PR TITLE
[CDE-825] - DashboardComponent: parameter propagation should take int…

### PR DESCRIPTION
**Pull request by eliofreitas**
…o account all the mapped parameters

### CODE REVIEW

**Comment by eliofreitas** [comment]
@pamval please review

**Comment by pamval** [comment]
@jvelasques @afrjorge please review

**Comment by eliofreitas** [comment]
@afrjorge @jvelasques updated

**Comment by eliofreitas** [comment]
Building pull request

**Comment by afrjorge** [comment]
Seems fine, @jvelasques agree?

File: cde-pentaho5/test-js/components/DashboardComponent-spec.js [hunk 1]
```diff
@@ -166,6 +167,45 @@ define([
     });
 
     /**
+     * ## The Dashboard Component # should propagate all Paramaters when one parameter changes from the main dashboard to the required dashboard
```
Comment (jvelasques) [inline cde-pentaho5/test-js/components/DashboardComponent-spec.js]: parameters

File: cde-pentaho5/test-js/components/DashboardComponent-spec.js [hunk 2]
```diff
@@ -166,6 +167,45 @@ define([
...
+        dashboard.setParameter("param1", "custom");
+        dashboard.fireChange("param3", "test");
+       
+      });
+
+      dashboard.update(dashboardComponent);
+    });
+    
+    /**
+     * ## The Dashboard Component # should propagate all Paramaters when one parameter changes from required dashboard to the main dashboard
```
Comment (jvelasques) [inline cde-pentaho5/test-js/components/DashboardComponent-spec.js]: parameters

**Comment by eliofreitas** [comment]
@afrjorge @jvelasques updated

**Comment by jvelasques** [comment]
@pamval @afrjorge looks good

File: cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js [hunk 3]
```diff
@@ -79,13 +79,19 @@ define([
               var eventName = myParam + ":fireChange";
               var fun = function(evt) {
                 if(reqDash.getParameterValue(otherParam) !== evt.value) {
-                  reqDash.fireChange(otherParam, evt.value);
+                   myself.loopThroughMapping(function(myParam, otherParam) {
+                     reqDash.setParameter(otherParam, myself.dashboard.getParameterValue(myParam));
+                   });
+                 reqDash.fireChange(otherParam, evt.value);
```
Comment (afrjorge) [inline cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js]: just to make it perfect, can you fix the indentation in the previous 4 lines?

File: cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js [hunk 4]
```diff
@@ -79,13 +79,19 @@ define([
...
                 }
               };
               myself.dashboard.on(eventName, fun);
               reqDash.on(otherParam + ":fireChange", function (evt) {
                 if((myself.oneWayMap == false) && (myself.dashboard.getParameterValue(myParam) !== evt.value)) {
-                  myself.dashboard.fireChange(myParam, evt.value);
+                   myself.loopThroughMapping(function(myParam, otherParam) {
+                     myself.dashboard.setParameter(myParam, reqDash.getParameterValue(otherParam));
+                   });
+                 myself.dashboard.fireChange(myParam, evt.value);
```
Comment (afrjorge) [inline cde-pentaho5/resource/resources/custom/amd-components/Dashboard/DashboardComponent.js]: same here, indentation in the previous 4 lines

**Comment by eliofreitas** [comment]
@updated

**Comment by pamval** [comment]
@jvelasques @afrjorge review please

**Comment by jvelasques** [comment]
looks good

**Pull request merged**